plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestVarcharToIntegralNumericCoercer.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.trino.plugin.hive.coercions.CoercionUtils.CoercionContext;
+import io.trino.spi.block.Block;
+import io.trino.spi.type.Type;
+import org.junit.jupiter.api.Test;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.plugin.hive.HiveTimestampPrecision.DEFAULT_PRECISION;
+import static io.trino.plugin.hive.HiveType.toHiveType;
+import static io.trino.plugin.hive.coercions.CoercionUtils.createCoercer;
+import static io.trino.spi.block.TestingSession.SESSION;
+import static io.trino.spi.predicate.Utils.nativeValueToBlock;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestVarcharToIntegralNumericCoercer
+{
+    @Test
+    public void testVarcharToTinyintCoercions()
+    {
+        assertVarcharToIntegralCoercion("0", TINYINT, (byte) 0);
+        assertVarcharToIntegralCoercion("-0", TINYINT, (byte) 0);
+        assertVarcharToIntegralCoercion("-1", TINYINT, (byte) -1);
+        assertVarcharToIntegralCoercion("+1", TINYINT, (byte) 1);
+        assertVarcharToIntegralCoercion("127", TINYINT, (byte) 127);
+        assertVarcharToIntegralCoercion("-128", TINYINT, (byte) -128);
+        assertVarcharToIntegralCoercion("128", TINYINT, (byte) 128); // Greater than Byte.MAX_VALUE
+        assertVarcharToIntegralCoercion("-129", TINYINT, (byte) -129); // Lesser than Byte.MIN_VALUE
+        assertVarcharToIntegralCoercion("2147483647", TINYINT, (byte) 2147483647);
```

Comment:
why expected output is `2147483647` for this case? shouldn't this is out of range for tinyint datatype?

Reply:
In case of Non-ORC coercions, Hive tries to get the integer representation of the value and cast them to the corresponding data type - So it is not exactly 2147483647 but casted to byte